loopin.go

```diff
@@ -89,10 +100,10 @@ func newLoopInSwap(globalCtx context.Context, cfg *swapConfig,
 	// route hints.
 	quote, err := cfg.server.GetLoopInQuote(
 		globalCtx, request.Amount, cfg.lnd.NodePubkey, request.LastHop,
-		nil,
+		routeHints,
```

Comment:
In addition to adding these hints to the quote, we also need to add them to the swap and probe invoices below so that we use the same set of hints for all operations. 

You'll need to make another lndclient PR to add these params, then update `go.mod` to point to the new version.

Reply:
Are you sure the swap was actually paid on a private channel? If you have private and public channels then the swap will succeed on the public channel. I can't get a swap to succeed here with a single private channel.

Reply:
Do you have more than 1 hop between primary and client? Because I'm working with the following setup and I can't get a success:

```
alice --(private)-- gateway --- primary
```

I don't see how the primary can make a payment for a swap to alice without hop hints, eg even if I try: 
```
(alice) lncli addinvoice --amt=10 --private=false
(primary) lncli sendpayment --pay_req={alice pay req}
```
(here, I set `--private=false` because the cli defaults to true, but the direct api calls that loop makes won't have that default)

But that same invoice can be paid by alice's direct peer because it knows about the channel. This is def something we should have client side itests for :/ much easier to try things out with that structure.

Reply:
@arshbot any input here?

Reply:
Why can't we just use lnd's existing logic and provide it with private channels as input? Since you've added that parameter in lightninglabs/lndclient/pull/66. It's ok if this doesn't succeed for +0 depth private channels, we can't route to that anyway.

> mainly due to another bug that got in the way

Please will you open an issue with steps to reproduce + description of problem if there's a bug?